resources/adoptopenjdk_install.rb

```diff
@@ -2,110 +2,91 @@
-    if adoptopendjk_downloaded?(download_path, new_resource)
-      Chef::Log.debug('AdoptOpenJDK tarball already downloaded, not downloading again')
-    else
-      Chef::Log.debug("downloading tarball from #{URI.parse(new_resource.url).host}")
-      remote_file "#{node['java']['download_path']}/#{tarball_name}" do
-        source new_resource.url
-        checksum new_resource.checksum
-        retries new_resource.retries
-        retry_delay new_resource.retry_delay
-        mode '644'
-        action :nothing
-      end.run_action(:create_if_missing)
-    end
-
-    converge_by("extract compressed data into Chef file cache path and move extracted data to #{app_dir}") do
-      package 'tar' do
-        not_if { platform_family?('mac_os_x', 'windows') }
-        action :nothing
-      end.run_action(:install)
-
-      cmd = shell_out(%(tar xvzf "#{node['java']['download_path']}/#{tarball_name}" -C "#{node['java']['download_path']}" --no-same-owner)
-                     )
-      unless cmd.exitstatus == 0
-        Chef::Application.fatal!("Failed to extract file #{tarball_name}!")
-      end
-
-      cmd = shell_out(
-        %(mv "#{node['java']['download_path']}/#{app_dir_name}" "#{app_dir}" )
-      )
-      unless cmd.exitstatus == 0
-        Chef::Application.fatal!(%( Command \' mv "#{node['java']['download_path']}/#{app_dir_name}" "#{app_dir}" \' failed ))
-      end
+  end
 
-      # change ownership of extracted files
-      FileUtils.chown_R new_resource.owner, app_group, app_dir
-    end
+  remote_file "#{Chef::Config[:file_cache_path]}/#{tarball_name}" do
+    source new_resource.url
+    checksum new_resource.checksum
+    retries new_resource.retries
+    retry_delay new_resource.retry_delay
+    mode '644'
   end
 
-  # set up .jinfo file for update-java-alternatives
-  java_name, jinfo_file = alternatives_config_file(app_root, app_home)
-  if platform_family?('debian') && !::File.exist?(jinfo_file)
-    converge_by("Add #{jinfo_file} for debian") do
-      template jinfo_file do
-        cookbook 'java'
-        source 'adoptopenjdk.jinfo.erb'
-        owner new_resource.owner
-        group app_group
-        variables(
-          priority: new_resource.alternatives_priority,
-          bin_cmds: new_resource.bin_cmds,
-          name: java_name,
-          app_dir: app_home
-        )
-        action :nothing
-      end.run_action(:create)
-    end
+  archive_file "#{Chef::Config[:file_cache_path]}/#{tarball_name}" do
+    destination extract_dir
   end
 
-  # link app_home to app_dir
-  Chef::Log.debug "app_home is #{app_home} and app_dir is #{app_dir}"
-  current_link = ::File.symlink?(app_home) ? ::File.readlink(app_home) : nil
-  if current_link != app_dir
-    converge_by("symlink #{app_dir} to #{app_home}") do
-      FileUtils.rm_f app_home
-      FileUtils.ln_sf app_dir, app_home
-      FileUtils.chown new_resource.owner, app_group, app_home
-    end
+  node.default['java']['java_home'] = new_resource.java_home
```

Comment:
why?

Reply:
So for now, we can pull it into another resource. This will be gone when I do a full removal of attributes in a few refactors time